axum-macros/src/typed_path.rs

```diff
@@ -18,52 +18,78 @@ pub(crate) fn expand(item_struct: ItemStruct) -> syn::Result<TokenStream> {
         ));
     }
 
-    let Attrs { path } = parse_attrs(attrs)?;
+    let Attrs { path, rejection } = parse_attrs(attrs)?;
 
     match fields {
         syn::Fields::Named(_) => {
             let segments = parse_path(&path)?;
-            Ok(expand_named_fields(ident, path, &segments))
+            Ok(expand_named_fields(ident, path, &segments, rejection))
         }
         syn::Fields::Unnamed(fields) => {
             let segments = parse_path(&path)?;
-            expand_unnamed_fields(fields, ident, path, &segments)
+            expand_unnamed_fields(fields, ident, path, &segments, rejection)
         }
-        syn::Fields::Unit => expand_unit_fields(ident, path),
+        syn::Fields::Unit => expand_unit_fields(ident, path, rejection),
     }
 }
 
+mod kw {
+    syn::custom_keyword!(rejection);
+}
+
 struct Attrs {
     path: LitStr,
+    rejection: Option<syn::Path>,
+}
+
+impl Parse for Attrs {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        let path = input.parse()?;
+
+        let _ = input.parse::<Token![,]>();
+
+        let rejection = if input.parse::<kw::rejection>().is_ok() {
```

Comment:
This `is_ok()` failing could mean both 'end of input' and 'found another token'. I think you should use `input.lookahead1()`. Also it's a bit weird to require the path to come first..

Reply:
You mean like this d289e7f?

> Also it's a bit weird to require the path to come first..

Why? I like that if you only specify the path you just do `#[typed_path("/foo")]`. I think thats the common case so nice that its short.